scipy/io/tests/test_mmio.py

```diff
@@ -387,6 +387,20 @@ def test_sparse_formats(self):
                 result = mmread(fn).todense()
                 assert_array_almost_equal(result, expected)
 
+    def test_precision(self):
+        for precision in range(1, 8):
+            # construct sparse matrix with pi as last main diagonal entry
+            n = 10**precision + 1
+            A = scipy.sparse.lil_matrix((n, n))
+            A[n-1, n-1] = pi
+            # write matrix with low precision and read again
+            mmwrite(self.fn, A, precision=precision)
+            A = scipy.io.mmread(self.fn)
+            # check for right entries in matrix
+            assert_array_equal(A.row, [n-1])
+            assert_array_equal(A.col, [n-1])
+            assert_array_almost_equal(A.data, [pi], decimal=precision-1)
```

Comment:
You are right. In your reference, `g` is also used.

But as a user, I would associate `precision` with decimal places. For example, if the general format `g` is used with `precision = 2`, `10**(-4)` would result in `0.0001`. This is not what I would intuitively expect.